manifests/guestcluster/1.31/pvcsi.yaml

```diff
@@ -0,0 +1,712 @@
+        - name: X_CSI_SPEC_REQ_VALIDATION
+          value: "false"
+        - name: CLUSTER_FLAVOR
+          value: "GUEST_CLUSTER"
+        - name: LOGGER_LEVEL
+          value: "PRODUCTION" # Options: DEVELOPMENT, PRODUCTION
+        - name: X_CSI_LOG_LEVEL
+          value: DEBUG
+        - name: CSI_NAMESPACE
+          value: {{ .PVCSINamespace }}
+        volumeMounts:
+        - name: plugin-dir
+          mountPath: 'C:\csi'
+        - name: pods-mount-dir
+          mountPath: 'C:\var\lib\kubelet'
+        - name: csi-proxy-volume-v1
+          mountPath: \\.\pipe\csi-proxy-volume-v1
+        - name: csi-proxy-filesystem-v1
+          mountPath: \\.\pipe\csi-proxy-filesystem-v1
+        - name: csi-proxy-disk-v1
+          mountPath: \\.\pipe\csi-proxy-disk-v1
+        - name: csi-proxy-system-v1alpha1
+          mountPath: \\.\pipe\csi-proxy-system-v1alpha1
+      - name: liveness-probe
+        image: registry.k8s.io/sig-storage/livenessprobe-win:v2.12.1
+        args:
+        - --csi-address=/csi/csi.sock
+        imagePullPolicy: "IfNotPresent"
+        volumeMounts:
+        - name: plugin-dir
+          mountPath: /csi
+      volumes:
+      - name: registration-dir
+        hostPath:
+          path: 'C:\var\lib\kubelet\plugins_registry\'
+          type: Directory
+      - name: plugin-dir
+        hostPath:
+          path: 'C:\var\lib\kubelet\plugins\csi.vsphere.vmware.com\'
+          type: DirectoryOrCreate
+      - name: pods-mount-dir
+        hostPath:
+          path: \var\lib\kubelet
+          type: Directory
+      - name: csi-proxy-disk-v1
+        hostPath:
+          path: \\.\pipe\csi-proxy-disk-v1
+          type: ''
+      - name: csi-proxy-volume-v1
+        hostPath:
+          path: \\.\pipe\csi-proxy-volume-v1
+          type: ''
+      - name: csi-proxy-filesystem-v1
+        hostPath:
+          path: \\.\pipe\csi-proxy-filesystem-v1
+          type: ''
+      - name: csi-proxy-system-v1alpha1
+        hostPath:
+          path: \\.\pipe\csi-proxy-system-v1alpha1
+          type: ''
+      tolerations:
+        - effect: NoExecute
+          operator: Exists
+        - effect: NoSchedule
+          operator: Exists
+---
+apiVersion: v1
+data:
+  cns-csi.conf: |
+    [GC]
+    endpoint = "{{ .SupervisorMasterEndpointHostName }}"
+    port = "{{ .SupervisorMasterPort }}"
+    tanzukubernetescluster-uid = "{{ .TanzuKubernetesClusterUID }}"
+    tanzukubernetescluster-name = "{{ .TanzuKubernetesClusterName }}"
+kind: ConfigMap
+metadata:
+  name: pvcsi-config
+  namespace: {{ .PVCSINamespace }}
+---
+apiVersion: v1
+data:
+  "volume-extend": "true"
+  "volume-health": "true"
+  "online-volume-extend": "true"
+  "file-volume": "true"
+  "csi-sv-feature-states-replication": "false" # Do not enable for guest cluster, Refer PR#2386 for details
+  "block-volume-snapshot": "true"
+  "tkgs-ha": "true"
+  "cnsmgr-suspend-create-volume": "true"
+  "csi-windows-support": "true"
+  "workload-domain-isolation": "true"
```

Comment:
not really sure if workload-domain-isolation should be true in 1.31 as well?

Reply:
Yes, this is needed for 1.31 as well